src/ensembl/src/shared/button/Button.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { ReactNode } from 'react';
+import classNames from 'classnames';
+
+import styles from './Button.scss';
+
+type Props = {
+  onClick: () => void;
+  isDisabled?: boolean;
+  className?: string;
+  children: ReactNode;
+};
+
+export const PrimaryButton = (props: Props) => {
+  const className = classNames(
+    styles.primaryButton,
+    { [styles.primaryButtonDisabled]: props.isDisabled },
+    props.className
+  );
+  return <Button {...props} className={className} />;
```

Comment:
Should we separate out `className` from props before passing it down to Button?

Reply:
It should overwrite the `className` in props, it that's what you are asking.

Reply:
I was wondering if it would be better to do something like `const {className, ...rest} = props` so that there is no overwriting?

Also, would `<Button  className={className} {...props} />` still work the same way?

Reply:
> would <Button className={className} {...props} /> still work the same way?

It shouldn't. [Later attributes override previous ones.](https://gist.github.com/sebmarkbage/07bbe37bc42b6d4aef81#spread-attributes).

I don't know whether explicitly destructuring `className` from the `props` will be better. It lacks explicitness, but loses terseness, and for a developer familiar with JSX it's apparent what's going on anyway. I can re-write it this way, if that's what you would prefer.

Reply:
Although it's apparent, I was just wondering if all the developers who will be working on updating this component in the future will be aware that this ordering cannot be changed.

Reply:
I would say that, since it's standard JSX behaviour, they will (or should, or eventually will) be aware of that. But as a safeguard, there already is a test in place that checks that classes are correctly assigned. It fails if you change the order of props:

![image](https://user-images.githubusercontent.com/6834224/55399047-f7caa980-5541-11e9-89e6-ea1b7e6bc591.png)

Reply:
That's cool, we have something to safeguard.